### PR TITLE
chore: update .gitignore and dependecies

### PR DESCRIPTION
**Pull request by Nictheboy**
Changed some configuration to make it easy to install the dependencies
in some strange environment, e.g. Windows or some cloud ide.

### CODE REVIEW

**Comment by Nictheboy** [comment]
Request a merge.

File: src/users/users.service.ts [hunk 1]
```diff
@@ -10,7 +10,7 @@
 import { Injectable, Logger } from '@nestjs/common';
 import { TokenExpiredError } from '@nestjs/jwt';
 import { InjectRepository } from '@nestjs/typeorm';
-import * as bcrypt from 'bcrypt';
+import * as bcrypt from 'bcryptjs';
```
Comment (andylizf) [inline src/users/users.service.ts:13]: why this change?

**Review by andylizf — APPROVED** [review]

File: src/users/users.service.ts (hunk 1 above)
Comment (Nictheboy) [reply src/users/users.service.ts:13]: bcrypt is implemented as a c++ native library, and there is compatibility issue when installing on some platforms. bcryptjs is completely in js and won't cause these problems.
